subsys/random/Kconfig

```diff
@@ -75,6 +75,19 @@ config XOSHIRO_RANDOM_GENERATOR
 
 endchoice # RNG_GENERATOR_CHOICE
 
+
+DT_CHOSEN_Z_ENTROPY := zephyr,entropy
+config CSPRNG_AVAILABLE
+	bool
+	default y if $(dt_chosen_enabled,$(DT_CHOSEN_Z_ENTROPY))
```

Comment:
Cleanliness-question:

Should we touch the device-tree here or should there be a generation of a prompt-less symbol called `DT_HAS_ENTROPY_ENABLED` somewhere in the system, instead?

I did a search for dt_chosen_enabled in Kconfigs (`zephyr/**/Kconfig*`) and I only find 19 entries... I find 3700 hits for `DT_HAS_XXXX_ENABLED` in Kconfig files

Reply:
Checking the `build/zephyr/include/generated/zephyr/autoconf.h` header file that is generated during the build and I did not find any `DT_HAS` for the chosen symbols.

For example:
`west build -b nrf52840dk/nrf52840 samples/hello_world/`
has
- `zephyr,entropy = &rng;` where `rng: random@4000d000` has status "okay" in `zephyr/build/zephyr/zephyr.dts` and
- `CONFIG_DT_HAS_NORDIC_NRF_RNG_ENABLED=y` in `build/zephyr/include/generated/zephyr/autoconf.h`

but no reference to chosen devicetree chosen property.

If we want to use DT_HAS I can modify `CSPRNG_AVAILABLE` as follows:
```
config CSPRNG_AVAILABLE
     bool
     default y
     depends on CONFIG_DT_HAS_NORDIC_NRF_RNG_ENABLED ||
                         CONFIG_DT_HAS_...._ENABLED ||
                         etc etc
```
I think the result would be the same :)

Reply:
@frkv The `DT_HAS_XXXX_ENABLED` symbols are automatically derived from the compat strings of DT nodes which are enabled. This is not related to the chosen node, and there's also no single compat string for entropy drivers (they all have their own). It's a common convention to set driver Kconfig dependencies on the existence of an enabled node with the right compat string, which is why you're seeing all of those occurrences, but I don't think it helps us in this situation since this is about a property on the chosen node.

Reply:
Thanks for the explanation @jhedberg. This is only nitpicking from me in the current PR at this point and I'm sure it can be considered for follow-up work if there needs to be a generic signal, or if we need to do larger structural changes to simplify things even further....